.github/workflows/translation-updates-to-main-repo.yml

```diff
@@ -0,0 +1,144 @@
+name: Work on main branch to pull new & updated translations from tf branch
+# Run on main branch
+
+on:
+  push:
+    branches:
+      - 'translation-files'
+    paths:
+      - contribute-pages/**
+      - documentation-pages/**
+      - general-pages/**
+      - general-strings/**
+  # Enable running this workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+      languages:
+        description: "Languages to pull (if none provided the site's config-file will be used)"
+        required: false
+        type: string
+
+jobs:
+  update-PO:
+    name: Generate & push translated files
+    runs-on: ubuntu-latest
+    steps:
+        # Check out the repository and download it to the runner, allowing to run actions against the code
+      - name: Get main repository
+        uses: actions/checkout@v2
+        with:
+          ref: master
+      - name: Get translation repository
+        uses: actions/checkout@v2
+        with:
+          ref: translation-files
+          path: translation-files
+      - run: git branch
+      - run: ls -al
+      - name: Install mdpo
+        run: pip install mdpo
+      - name: Set languages
+        id: lang-set
+        run: |
+          if [[ -n "${{ github.event.inputs.languages }}" ]]; then
+            languagelist=${{ github.event.inputs.languages }}
+          else
+            languagelist=$(cat _config.yml | grep "languages: \[.*\]" )
+            languagelist=${languagelist//[}
+            languagelist=${languagelist//]}
+            echo $languagelist
+            languagelist=${languagelist//'languages: '}
+          fi
+          languagelist=${languagelist//,}
+          languagelist=${languagelist//;}
+          # Exclude English (source) from list of languages to update
+          languagelist=${languagelist//'en'}
+          # Make variable consistent between steps
+          echo "languages=$languagelist" >> $GITHUB_ENV
+      - name: Print tree
+        run: |
+          sudo apt-get install tree
```

Comment:
I would install things all at the beginning. Then the command output does not get mixed with installation output.